src/resource/resource.ts

```diff
@@ -273,8 +279,48 @@ export class FetchError extends SolidClientError {
     return this.response.statusText;
   }
 
+  async problemDetails(): Promise<ProblemDetails> {
```

Comment:
This is different to the `problemDetails()` function in: https://github.com/inrupt/solid-client-errors-js/blob/main/src/http/httpError.ts#L83. This difference will mean clients will have to handle errors differently.

This should be `get problemDetails()` so a client can call it like this `error.problemDetails`. Also it should not be async.